src/swarm/neo/request/RequestEventDispatcher.d

```diff
@@ -701,6 +681,27 @@ public struct RequestEventDispatcher
         enableStomping(this.waiting_fibers);
     }
 
+    /***************************************************************************
+
+        Unregisters all events for `fiber`.
+
+        Params:
+            fiber = fiber to unregister events for
+
+    ***************************************************************************/
+
+    private void unregisterFiber ( MessageFiber fiber )
+    {
+        this.waiting_fibers.length = Array.moveToEnd(this.waiting_fibers,
```

Comment:
`removeShift` is not suitable here?

Reply:
`removeShift` removes a single element, whereas here we want to remove several matching elements in one pass.

Reply:
(The 2nd commit refactors this method again, anyway.)

Reply:
> removeShift removes a single element, whereas here we want to remove several matching elements in one pass.

Ah :+1: